src/classes/STG_PanelHealthCheck_CTRL.cls

```diff
@@ -62,6 +62,45 @@ public without sharing class STG_PanelHealthCheck_CTRL extends STG_Panel {
         } private set;
     }
 
+    /*********************************************************************************************************
+    * @description Override saveSettings from STG_Panel to update Custom Setting Error Setting
+    *   DisableRecordDataHealthChecks__c field, which allows user to only run config related health checks
+    * @return PageReference 
+    */
+    public override PageReference saveSettings() {
+        STG_SettingsManager_CTRL.idPanelCurrent = idPanel();
+        Savepoint sp = Database.setSavepoint();
+        try {
+            Boolean hasUpdatePermission = UTIL_Describe.getObjectDescribe(
+                String.valueOf(Error_Settings__c.SObjectType)
+            ).isUpdateable();
+
+            if (!hasUpdatePermission) {
+                throw new NoPermissionException(
+                    String.format(
+                        System.Label.flsError,
+                        new List<String>{Schema.SObjectType.Error_Settings__c.getLabel()})
+                );
+            }
+
+            if (STG_Panel.stgService.stgErr != null && !Test.isRunningTest()) {
+                update STG_Panel.stgService.stgErr;
+            }
+
+            // Health Checks reulst will be clear out when Error Setting DisableRecordDataHealthChecks__c field
+            // is successfully updated because the result will not be relevant anymore.
+            listDR = new List<DetectResult>();
+
+        } catch (Exception e) {
+            Database.rollback(sp);
+            ERR_Handler.processError(e, ERR_Handler_API.Context.STTG);
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR, e.getMessage()));
+        }
+
+        return null;
+    }
+
+    public class NoPermissionException extends Exception {}
```

Comment:
`NoPermissionException` -> `HealthCheckException` since we can generate other exceptions that are not related to permission checks.